src/impl/unicast/transport/NServiceBus.Unicast.Transport.Transactional.Config/AdvancedTransactionalConfig.cs

```diff
@@ -10,8 +15,39 @@ public static class AdvancedTransactionalConfig
         /// <returns></returns>
         public static Configure SupressDTC(this Configure config)
         {
-            Bootstrapper.SupressDTC = true;
+            if (UserSuppressedOutdatedDispatchWarning())
+            {
+                Bootstrapper.SupressDTC = true;
+                return config;
+            }
+
+            var exceptionMessage = string.Format(
+                "You are using an outdated timeout dispatch implementation which can lead to message loss! " +
+                "Please update NServiceBus package to version 4.4.8 or higher. " +
+                "You can suppress this warning by calling 'SuppressOutdatedTimeoutDispatchWarning()' before 'SuppressDTC()' (e.g. 'configure.SuppressOutdatedTimeoutDispatchWarning().SupressDTC()')" +
+                " or by adding key '{0}' with value 'true' to the appSettings section of your application configuration file.", SuppressOutdatedTimeoutDispatchSetting);
+            throw new Exception(exceptionMessage);

```

Comment:
was just some thought to avoid throwing a generic Exception (which normally requires the user to catch the generic Exception). But since catching and handling that exception won't help the user anyway I have no strong opinion on that anyway.